conan/tools/layout/__init__.py

```diff
@@ -0,0 +1,79 @@
+import os
+
+from conans.client.file_copier import FileCopier
+from conans.errors import ConanException
+
+
+# FIXME: Not documented, just a POC, missing a way to do something similar without modifying
+#        the recipe
+
+def clion_layout(conanfile):
+    if not conanfile.settings.get_safe("build_type"):
+        raise ConanException("The 'clion_layout' requires the 'build_type' setting")
+    base = "cmake-build-{}".format(str(conanfile.settings.build_type).lower())
+    conanfile.folders.build.folder = base
+    conanfile.build_cpp_info.libdirs = ["."]
+
+    conanfile.folders.generators.folder = os.path.join(base, "generators")
+
+    conanfile.folders.source.folder = "."
+    conanfile.source_cpp_info.includedirs = ["include"]
+
+
+# FIXME: Not sure about the location, interface, name etc.
+#        Do we want a public explicit FileCopier maybe? Review and suggest!
```

Comment:
Yes, I have been wanting to make ``FileCopier`` explicit in recipes for a while. This seems related.